components/utilities/chartLayout.js

```diff
@@ -149,9 +149,9 @@
             chartElements.chartArea = chart;
 
             // Defs - for clipping path
-            var defs = chart.selectAll('defs').data(dummyData);
-            defs.enter().append('defs');
-            defs.exit().remove();
+            chartElements.defs = chart.selectAll('defs').data(dummyData);
+            chartElements.defs.enter().append('defs');
+            chartElements.defs.exit().remove();
```

Comment:
I agree, I'm not keen on the repetition, however, the above isn't equivalent - `enter()`, `append()` and `exit()` (although `exit()` is not relevant in this case) return different selections, so chaining the methods won't work.

When using both `enter()` _and_ `exit()`, I think the repetition is necessary, otherwise you'd need to perform the selection and data-join again; it's also consistent with the other components and is what you tend to see with [data-joins](http://bost.ocks.org/mike/join/) and the [general update pattern](http://bl.ocks.org/mbostock/3808234).

Would something like this be more agreeable stylistically? I think it's easier to read.

```js
var defs = chart.selectAll('defs').data(dummyData);
defs.enter().append('defs');
defs.exit().remove();

chartElements.defs = defs;
```